tests/general/base.py

```diff
@@ -0,0 +1,200 @@
+            )
+        return pcs
+
+    def create_dataset(self, profile, version):
+        return DatasetFactory(
+            profile=profile, version=version
+        )
+
+    def create_datasetdata(
+            self, dataset, geography, data={},
+            create_groups=True
+        ):
+        obj = DatasetDataFactory(
+            dataset=dataset, geography=geography, data=data
+        )
+
+        if create_groups:
+            keys = data.keys()
+
+            for key in keys:
+                if key == "count":
+                    continue
+
+                group = Group.objects.filter(dataset=dataset, name=key).first()
+                if not group:
+                    group = GroupFactory(
+                        dataset=dataset, name="gender", subindicators=[],
+                        can_aggregate=True, can_filter=True
+                    )
+
+                if data[key] not in group.subindicators:
+                    subindicators = group.subindicators
+                    subindicators.append(data[key])
+                    group.subindicators = subindicators
+                    group.save()
+        return obj
+
+    def multiple_datasetdata(self, dataset, geography, datalist=[]):
+        objs = []
+        for data in datalist:
+            objs.append(self.create_datasetdata(dataset, geography, data))
+        return objs
+
+    def create_indicator(self, dataset, group):
+        indicator = IndicatorFactory(dataset=dataset, groups=[group])
+        indicator_data_extraction(indicator)
+        return indicator
+
+
+    def create_highlight(
+            self, profile, indicator, subindicator,
+            label="highlight",
+            denominator="absolute_value"
+        ):
+        subindicator_index = indicator.subindicators.index(subindicator)
+        return ProfileHighlightFactory(
+            profile=profile, indicator=indicator,
+            subindicator=subindicator_index, denominator=denominator,
+            label=label
+        )
+
+    def create_key_metrics(
+            self, profile, indicator, subindicator,
+            label="metrics", denominator="absolute_value"
+        ):
+        subindicator_index = indicator.subindicators.index(subindicator)
+        category = IndicatorCategoryFactory(
+            profile=profile, name=f'category-{subindicator}'
+        )
+        subcategory = IndicatorSubcategoryFactory(
+            category=category, name=f'subcategory-{subindicator}'
+        )
+        return ProfileKeyMetricsFactory(
+            profile=profile, variable=indicator,
+            subindicator=subindicator_index, denominator=denominator,
+            label=label, subcategory=subcategory
+        )
+
+    def create_base_data(
+            self, version=None, geography=None,
+            hierarchy=None, profile=None,
+            create_version=True, create_boundary=True
+        ):
+
+        if create_version:
+            version = VersionFactory(name=version)
```

Comment:
it isn't nice to use the same variable name for two different types. It gets confusing and introduces bugs. 
```suggestion
            version = VersionFactory(name=version_name)
```

Reply:
Done